intelmq/bots/experts/sieve/README.md

```diff
@@ -0,0 +1,166 @@
+  add! comment = 'domainabuse'
+  keep
+} elsif classification.type == 'scanner' {
+  add! comment = 'ignore'
+  drop
+} else {
+  remove comment
+}
+```
+
+
+## Parameters
+
+The sieve bot takes only one parameter:
+ * `file` - filesystem path the the sieve file
+
+
+## Reference
+
+### Sieve File Structure
+
+The sieve file contains an arbitrary number of rules of the form:
+
+```
+if EXPRESSION {
+    ACTIONS
+} elif EXPRESSION {
+    ACTIONS
+} else {
+    ACTIONS
+}
+```
+
+
+###  Expressions
+
+Each rule specifies on or more expressions to match an event based on its keys
+and values. Event keys are specified as strings without quotes. String values
+must be enclosed in single quotes. Numeric values can be specified as integers
+or floats and are unquoted. IP addresses and network ranges (IPv4 and IPv6) are
+specified with quotes. Following operators may be used to match events:
+
+ * `:exists` and `:notexists` match if a given key exists, for example:
+
+    ```if :exists source.fqdn { ... }```
+
+ * `==` and `!=` match for equality of strings and numbers, for example:
+
+   ```if feed.name != 'acme-security' || feed.accuracy == 100 { ... }```
+
+ * `:contains` matches on substrings.
+
+ * `=~` matches strings based on the given regex. `!~` is the inverse regex
+ match.
+
+ * Numerical comparisons are evaluated with `<`, `<=`, `>`, `>=`.
+
+ * `<<` matches if an IP address is contained in the specified network range:
+
+   ```if source.ip << '10.0.0.0/8' { ... }```
+
+ * Values to match against can also be specified as list, in which case any one
+ of the values will result in a match:
+
+   ```if source.ip == ['8.8.8.8', '8.8.4.4'] { ... }```
+
+  In this case, the event will match if it contains a key `source.ip` with
+  either value `8.8.8.8` or `8.8.4.4`.
+
+
+### Actions
+
+If part of a rule matches the given conditions, the actions enclosed in `{` and
+`}` are applied. By default, all events that are matched or not matched by rules
+in the sieve file will be forwarded to the next bot in the pipeline, unless the
+`drop` action is applied.
+
+ * `add` adds a key value pair to the event. This action only applies if the key
+ is not yet defined in the event. If the key is already defined, the action is
+ ignored. Example:
+
+   ```add comment = 'hello, world'```
+
+ * `add!` same as above, but will force overwrite the key in the event.
+
+ * `modify` modifies an existing value for a key. Only applies if the key is
+already defined. If the key is not defined in the event, this action is ignored.
+Example:
+
+   ```modify feed.accuary = 50```
+
+ * `remove` removes a key/value from the event. Action is ignored if the key is
+ not defined in the event. Example:
+
+    ```remove extra.comments```
+
+ * `keep` marks the event to be forwarded to the next bot in the pipeline
+ (same as the default behaviour), but in addition the sieve file processing is
+ interrupted upon reaching this action.
```

Comment:
So implicitly here you are saying that the default behaviour without `keep` is to contineue with the matching ?

Reply:
Yes, the `keep` and `drop` actions explicitly interrupt the processing of the rules.

Without a mention of `keep` or `drop`, processing of subsequent rules continues. Afaik this is the same behavior as mail sieve.